new/rfcXXX-policyfile-endpoint.md

```diff
@@ -0,0 +1,485 @@
+contain a `dotted_decimal_identifier` field which contains a representation of
+the identifier as a version number.  This field is used when cookbooks are
+stored on a server that does not support the arbitrary cookbook identifier API
+described in Chef RFC022.
+
+Example cookbook locks including optional fields:
+
+```json
+    "policyfile_demo": {
+      "version": "0.1.0",
+      "identifier": "f04cc40faf628253fe7d9566d66a1733fb1afbe9",
+      "dotted_decimal_identifier": "67638399371010690.23642238397896298.25512023620585",
+      "source": "cookbooks/policyfile_demo",
+      "cache_key": null,
+      "scm_info": {
+        "scm": "git",
+        "remote": "git@github.com:danielsdeleo/policyfile-jenkins-demo.git",
+        "revision": "edd40c30c4e0ebb3658abde4620597597d2e9c17",
+        "working_tree_clean": false,
+        "published": false,
+        "synchronized_remote_branches": [
+
+        ]
+      },
+      "source_options": {
+        "path": "cookbooks/policyfile_demo"
+      }
+    },
+    "apt": {
+      "version": "2.6.1",
+      "identifier": "5f7045a8aeaf6ccda3b3594258df9ee982b3a023",
+      "dotted_decimal_identifier": "26863567272587116.57882360917678303.174725757378595",
+      "cache_key": "apt-2.6.1-supermarket.chef.io",
+      "origin": "https://supermarket.chef.io/api/v1/cookbooks/apt/versions/2.6.1/download",
+      "source_options": {
+        "artifactserver": "https://supermarket.chef.io/api/v1/cookbooks/apt/versions/2.6.1/download",
+        "version": "2.6.1"
+      }
+    },
+```
+
+#### Other Optional Fields
+
+A Policyfile lock can contain arbitrary information in other top-level fields.
+The Chef Server MUST accept and preserve the data in these fields (though the
+server MAY enforce a limit on overall document size). ChefDK currently uses a
+`solution_dependencies` field to store a list of all dependencies relevant to
+the cookbook set.
+
+### API Schema
+
+#### `/policy_groups`
+
+Container for policy groups.
+
+##### `GET /policy_groups`
+
+Returns a list of policy groups that exist.
+
+#### `/policy_groups/:policy_group_name/_acl`
+
+Authorization endpoint for the policy group `:policy_group_name`
+
+#### `/policy_groups/:policy_group_name/policies/`
+
+##### `GET /policy_groups/:policy_group_name/policies/`
+
+Returns a JSON object showing the active policy names and revision IDs for the
+policy group `:policy_group_name`.
+
+#### `/policy_groups/:policy_group_name/policies/:policy_name`
+
+##### `GET /policy_groups/:policy_group_name/policies/:policy_name`
+
+Returns the policyfile lock document at the revision that is currently
+associated with the given policy group and policy name.
+
+##### `PUT /policy_groups/:policy_group_name/policies/:policy_name`
+
+Sets the active revision of `:policy_name` for the policy group
+`:policy_group_name` to the policyfile lock document in the request body. The
+policy group, policy name, and policyfile lock revision will be created if any
+does not exist.
```

Comment:
via @manderson26 in chat, this section needs to clarify that existing revisions cannot be updated. Since this is the "do what I mean" endpoint, if a given revision already exists, the outcome of the request should be the same as `POST /policy_groups/:policy_group_name/policies/:policy_name` with the `revision_id` in the request body (see next endpoint description).